src/MongoDB.EntityFrameworkCore/Extensions/MongoPropertyBuilderExtensions.cs

```diff
@@ -53,65 +54,127 @@ public static class MongoPropertyBuilderExtensions
         => (PropertyBuilder<TProperty>)HasElementName((PropertyBuilder)propertyBuilder, name);
 
     /// <summary>
-    /// Configures the <see cref="DateTimeKind"/> for the property.
+    /// Configures the document element that the property is mapped to when targeting MongoDB.
+    /// If an empty string is supplied then the property will not be persisted.
     /// </summary>
     /// <param name="propertyBuilder">The builder for the property being configured.</param>
-    /// <param name="dateTimeKind">The <see cref="DateTimeKind"/> to use for the property.</param>
-    /// <returns>The same builder instance so that multiple calls can be chained.</returns>
-    public static PropertyBuilder HasDateTimeKind(
-        this PropertyBuilder propertyBuilder,
-        DateTimeKind dateTimeKind)
+    /// <param name="name">The element name for the property.</param>
+    /// <param name="fromDataAnnotation">Indicates whether the configuration was specified using a data annotation.</param>
+    /// <returns>The same builder instance if the configuration was applied, <see langword="null" /> otherwise.</returns>
+    public static IConventionPropertyBuilder? HasElementName(
+        this IConventionPropertyBuilder propertyBuilder,
+        string? name,
+        bool fromDataAnnotation = false)
     {
-        propertyBuilder.Metadata.SetDateTimeKind(dateTimeKind);
+        if (!CanSetElementName(propertyBuilder, name, fromDataAnnotation))
+        {
+            return null;
```

Comment:
Should we not be throwing an `InvalidOperationException` in this case? If we return `null`, then we will throw a `NullReferenceException` when we chain the fluent API.

Reply:
This is the pattern used across all EF providers. You can see it in the Cosmos, SqlServer and relational base providers.